File: frontend/src/types/company.ts
export interface CompanyRequest {
  name: string;
  description?: string;
  website?: string;
  industry?: string;
  location?: string;
  logoUrl?: string;
  careersPageUrl?: string;
}

export interface CompanyResponse {
  id: number;
  name: string;
  description?: string;
  website?: string;
  industry?: string;
  location?: string;
  logoUrl?: string;
  careersPageUrl?: string;
  userId?: number;
  createdById?: number;
  createdAt?: string;
}

File: backend/src/main/java/com/jobtracker/entity/Company.java
package com.jobtracker.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "companies")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Company {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    @Column(length = 500)
    private String description;
    
    @Column(length = 200)
    private String website;
    
    @Column(length = 100)
    private String industry;
    
    @Column(length = 100)
    private String location;
    
    @Column(name = "logo_url", length = 500)
    private String logoUrl;
    
    @Column(name = "careers_page_url", length = 500)
    private String careersPageUrl;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;
    
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
